src/lib.rs

```diff
@@ -224,6 +226,35 @@ impl Format {
     pub const EMPTY: Format = Format::Tuple(Vec::new());
 }
 
+pub struct FormatModule {
+    names: Vec<String>,
+    formats: Vec<Format>,
+}
+
+impl FormatModule {
+    pub fn new() -> FormatModule {
+        FormatModule {
+            names: Vec::new(),
+            formats: Vec::new(),
+        }
+    }
+
+    pub fn define_format(&mut self, name: impl Into<String>, format: Format) -> Format {
+        let level = self.names.len();
+        self.names.push(name.into());
+        self.formats.push(format);
+        Format::ItemVar(level)
```

Comment:
Interesting that this is called "level", suggests nesting? :thinking:

Reply:
Level in the sense of a de Bruijn level as opposed to a de Bruijn index. This is in contrast to `Expr::Var` which uses a de Bruijn index.